src/ansys/grantami/jobqueue/_models.py

```diff
@@ -138,61 +138,101 @@ def to_dict(self) -> Dict[str, Union[int, None]]:
         }
 
 
-class _JobFile:
+class JobFile:
     """
-    Represents a file associated with a job request.
+    Represents a file associated with a JobRequest.
 
     Parameters
     ----------
-    file_type : _FileType
-        Type of file being represented.
+    path : str or pathlib.Path
+        Path to the local file to use in a JobRequest. Can be absolute or relative.
+    virtual_path : str or pathlib.Path
+        Virtual path to use to refer to the file in the JobRequest. Must be relative.
     """
 
-    def __init__(self, file_type: _FileType):
-        self.file_type = file_type
-        self._path: Union[pathlib.Path, None] = None
-        self._id: Optional[str] = None
+    def __init__(self, path: Union[str, pathlib.Path], virtual_path: Union[str, pathlib.Path]):
+        self._path: pathlib.Path = pathlib.Path(path)
+        virtual_path = pathlib.Path(virtual_path)
+        self._validate_virtual_path(virtual_path)
+        self._virtual_path: pathlib.Path = virtual_path
 
-    @classmethod
-    def from_pathlib_path(cls, path: pathlib.Path, file_type: _FileType) -> "_JobFile":
+    @staticmethod
+    def _validate_virtual_path(path: pathlib.Path) -> None:
         """
-        Create a ``JobFile`` object from a ``pathlib.Path`` object.
+        Validate the provided virtual path. It must be relative and within the working directory.
 
         Parameters
         ----------
         path : pathlib.Path
-            Path to the file.
-        file_type : _FileType
-            Type of file to create.
+            Path to validate.
+        """
+        if path.is_absolute():
+            raise ValueError("Virtual path must be relative.")
+        if not path.resolve().is_relative_to(os.getcwd()):
```

Comment:
Surely a relative path is relative to any directory you care to test it against?

Reply:
This handles paths going out of the current directory, e.g. `../some/folder/text.txt`